hadoop-hdfs-project/hadoop-hdfs-rbf/src/test/java/org/apache/hadoop/hdfs/server/federation/router/TestRouterSafemode.java

```diff
@@ -141,6 +141,31 @@ public void testRouterExitSafemode()
     verifyRouter(RouterServiceState.RUNNING);
   }
 
+  @Test
+  public void testRouterExitSafemodeResetUpTime()
+      throws InterruptedException, IllegalStateException, IOException {
+
+    Calendar calendar = Calendar.getInstance();
+    // Get the future times, add one day to the current date.
+    calendar.add(Calendar.DAY_OF_MONTH, 1);
+    long timestampAfterOneDay = calendar.getTimeInMillis();
+    router.getSafemodeService().setStartupTime(timestampAfterOneDay);
+
+    assertTrue(router.getSafemodeService().isInSafeMode());
+    verifyRouter(RouterServiceState.SAFEMODE);
+
+    // Wait for initial time in milliseconds
```

Comment:
One tiny issue, please unify the comments pattern.  Add dot in the end of a sentence.

Reply:
Thanks @hfutatzhanghb help me reivew it, i wll update it later.